src/common/SmartDataSerialize.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "SmartData.h"
+#include <QDataStream>
+
+inline QDataStream& operator<<(QDataStream& _out, const SmartData::AttrData& _attrData)
```

Comment:
why _ in parameter names?

Reply:
I write in this style since this project started.